### DISCUSSION TITLE
Self hosting gatsby functions

### DISCUSSION THREAD

**Discussion by ZeldOcarina** [gatsbyjs/gatsby#34354] [Help]
Hi!
Is there a way to host gatsby functions outside Gatsby Cloud or Netlify? It's the only two services that support those.
I am trying to deploy a website with a Gatsby function (a single one btw) on Cloudflare pages and converting that to their function model is super laborious.
How would you host that in a VPS?
Any suggestions? Thank you!

**Comment by herecydev** [comment]
The netlify stuff is fairly transparent on how it can be done. So I would look at that, it's non trivial but if you're willing to put in the effort it's clearly doable.

**Reply by ZeldOcarina** [reply]
Hi @herecydev thanks for your answer. What I mean is hosting those OUTSIDE of those two services! :)

**Reply by herecydev** [reply]
Yer I got what you meant. But there are no guides so I'm telling you to engineer a solution by looking at netlify and how they accomplished it, learn from it and apply what you've learnt to cloudflare pages.
